operator/internal/engine/mysql_test.go

```
@@ -0,0 +1,199 @@
+	ua, ub := mysqlUserName(a), mysqlUserName(b)
+	if ua == ub {
+		t.Fatalf("mysqlUserName collided: %q and %q both gave %q", a, b, ua)
+	}
+	if len(ua) > mysqlMaxUserLength || len(ub) > mysqlMaxUserLength {
+		t.Fatalf("derived users exceed the limit: %q (%d), %q (%d)", ua, len(ua), ub, len(ub))
+	}
+}
+
+func TestMySQLUserNameIsDeterministic(t *testing.T) {
+	in := strings.Repeat("c", 63)
+	if mysqlUserName(in) != mysqlUserName(in) {
```

golangci-lint (2.12.2)

[error] SA4000: identical expressions on the left and right side of the '!=' operator

(staticcheck)